### DISCUSSION TITLE
customize FDM 3.x codes and make xdm

### DISCUSSION THREAD

**Discussion by plb7777** [subhra74/xdm#930] [Ideas]
Dear subhra 
To use dotnet requires dotnet framework to be installed in the pc to run the application. I think a newer approach would be a better idea for new xdman. i am explaining all the procedure step by step here kindly take a moment and carefully think about it.
1. FDM 3.x sourcecode is available in sourceforge as open source and using it to develop own software is not illegal.
2. FDM UI seems out of date but personally i find it more productive then modern UI (i have used xdm 4.7 more than other xdman).
3. The core downloading protocol is very much advanced in FDM 3.9 and it can support http, https, ftp, ftps, dash, rtsp, mms, hls etc all type of downloads.
Now the main question is how to do it. This could be done in three major steps one is updating/redesigning UI and secondly reforming core download engine and thirdly browser integration with download button support above web players.
Change in UI:
1. remove all the tabs present in main UI (remove the codes as well), all the downloads will be implemented in main download list view.
2. the menubar should be minimized to file, download, tools and help.
3. the toolbar should contain add url, parse url, start, pause, remove and settings buttons.
4. category view should be removed, category and sort will be available in download menu at menubar.
5. in the bottom statusbar only display link to social media site and downloaded size and file size should be displayed of selected download.
6. below downloadlist view logs, progress, media preview and opinions should be removed.
7. downloadlist per download contain status, filename, size, speed, eta, status and progressbar.
8. in settings all the less necessary settings will be removed and provided with most imp settings. there should be option to select yt-dlp/youtube-dl and ffmpeg.
9. in browser integration it should support all browser and show a download button above web players.
10. parse url will use third party apps (like yt-dlp or youtube-dl) to scan links and convert the data to be download using the download manager.

much more to come during development.
kindly make some time and think about this because like me many users are running low end pc and using modern download manager can cause system lag. fdm 3.x donot slow the system but lacks so much features. 

thanking you

**Comment by subhra74** [comment]
To run new XDM you do not need to install dotnet framework, It already comes with Windows 10. New XDM is also very lightweight. takes around 30mb of RAM. FDM was written in C++ which causes more issues in bug fixing and difficult to maintain for me. XDM already supports all browser. XDM also supports all modern protocols that are used in 2022. Give the new version a try, its still in beta but still should work pretty well
https://github.com/subhra74/xdm-experimental-binaries/releases/tag/8.0.18-beta

**Reply by numberZero** [reply]
Hi, why was I mentioned? (and was I? GitHub tells me so but I don’t see the actual mention)
![image](https://user-images.githubusercontent.com/7352626/211192877-ac822712-bc04-42e6-b700-917a73255aa9.png)

**Reply by Mantarri** [reply]
I had the same thing happen, and it seems I'm subscribed to the issue somehow. Just unsubscribe.
![image](https://user-images.githubusercontent.com/47439660/211214011-224ae74b-2658-4d4e-9247-e8d7e6464ac4.png)
